amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java

```diff
@@ -166,56 +202,60 @@ protected String getQuery()
     /**
      * @see org.alfresco.module.org_alfresco_module_rm.job.RecordsManagementJobExecuter#execute()
      */
+    @Override
     public void executeImpl()
     {
         try
         {
             logger.debug("Job Starting");
 
-            if (dispositionActions != null && !dispositionActions.isEmpty())
+            if (dispositionActions == null || dispositionActions.isEmpty())
             {
-                boolean hasMore = true;
-                int skipCount = 0;
+                logger.debug("Job Finished as disposition action is empty");
+                return;
+            }
 
-                if (batchSize < 1)
+            boolean hasMore = true;
+            int skipCount = 0;
+
+            if (batchSize < 1)
+            {
+                if (logger.isDebugEnabled())
                 {
-                    if (logger.isDebugEnabled())
-                    {
-                        logger.debug("Invalid value for batch size: " + batchSize + " default value used instead.");
-                    }
-                    batchSize = DEFAULT_BATCH_SIZE;
+                    logger.debug("Invalid value for batch size: " + batchSize + " default value used instead.");
                 }
-                if (logger.isTraceEnabled())
+                batchSize = DEFAULT_BATCH_SIZE;
+            }
+            if (logger.isTraceEnabled())
+            {
+                logger.trace("Using batch size of " + batchSize);
+            }
+
+            while (hasMore)
+            {
+                final SearchParameters params = new SearchParameters();
```

Comment:
I don't see the benefit of declaring this final (also applies to lots of other variables below).

Reply:
Making a variable final prevents the variable from being changed if it is expected to stay constant. It is good to declare members final if you do not intend to change the reference at all.

Reply:
I think it's better to avoid using `final` where ever possible, since it has no practical benefit (final objects are not really constants in Java since their state can be changed) and it detracts from readability.

I think it's worth using final for primitives and immutable objects (particularly for constants), but otherwise it just adds clutter to the code.